### PR TITLE
KIE forms jBPM integration

### PR DESCRIPTION
**Pull request by pefernan**
- Added FormProviding engine
- Cleaned up unused modules.
- Improved displayer APIs
- Improved forms rendering engine
- Fixing UI errors
- Added KieWorkbench form displayers
- Added generic start process UI

### CODE REVIEW

**Comment by pefernan** [comment]
@Salaboy @cristianonicolai @nmirasch can you review it?

File: jbpm-console-ng-generic/jbpm-console-ng-generic-forms/jbpm-console-ng-generic-forms-backend/src/main/java/org/jbpm/console/ng/ga/forms/display/backend/provider/FreemakerFormProvider.java [hunk 1]
```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright 2014 JBoss by Red Hat.
```
Comment (cristianonicolai) [inline jbpm-console-ng-generic/jbpm-console-ng-generic-forms/jbpm-console-ng-generic-forms-backend/src/main/java/org/jbpm/console/ng/ga/forms/display/backend/provider/FreemakerFormProvider.java]: 2016...

File: jbpm-console-ng-generic/jbpm-console-ng-generic-forms/jbpm-console-ng-generic-forms-backend/src/main/resources/forms/DefaultProcess.ftl [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+                    <h2>Outputs</h2><br/>
+                    <#list outputs?keys as key>
+                             <div class="form-row clearfix">
+                                <label>${key}</label>
+                                    <input type="text" name="${key}" value=""/>
+                            </div>
+
+                    </#list>
+                </#if>
+                <i>* Automatically generated form, only supports simple text strings *</i>
```
Comment (cristianonicolai) [inline jbpm-console-ng-generic/jbpm-console-ng-generic-forms/jbpm-console-ng-generic-forms-backend/src/main/resources/forms/DefaultProcess.ftl]: Is this message really necessary? If so, we need proper i18n handling.

File: jbpm-console-ng-generic/jbpm-console-ng-generic-forms/jbpm-console-ng-generic-forms-backend/src/main/resources/forms/DefaultTask.ftl [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+                             <div class="form-row clearfix">
+                                <label>${key}</label>
+                                <div class="input-button">
+                                    <input type="text" name="${key}" value="${outputs[key]!""}"/>
+                                </div>
+                            </div>
+                        </#if>
+                    </#list>
+                </#if>
+                <i>* Automatically generated form, only supports simple text strings *</i>
```
Comment (cristianonicolai) [inline jbpm-console-ng-generic/jbpm-console-ng-generic-forms/jbpm-console-ng-generic-forms-backend/src/main/resources/forms/DefaultTask.ftl]: Same here and for Input/Output labels...

File: jbpm-console-ng-human-tasks-forms-modeler/jbpm-console-ng-human-tasks-forms-modeler-backend/src/main/java/org/jbpm/console/ng/ht/forms/backend/server/finder/RuntimeFormFinder.java [hunk 4]
```diff
@@ -0,0 +1,108 @@
+/*
+ * Copyright 2015 Red Hat, Inc. and/or its affiliates.
```
Comment (cristianonicolai) [inline jbpm-console-ng-human-tasks-forms-modeler/jbpm-console-ng-human-tasks-forms-modeler-backend/src/main/java/org/jbpm/console/ng/ht/forms/backend/server/finder/RuntimeFormFinder.java]: 2016

File: jbpm-console-ng-human-tasks-forms-modeler/jbpm-console-ng-human-tasks-forms-modeler-backend/src/main/java/org/jbpm/console/ng/ht/forms/backend/server/impl/FormRenderContentMarshallerManagerImpl.java [hunk 5]
```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright 2015 Red Hat, Inc. and/or its affiliates.
```
Comment (cristianonicolai) [inline jbpm-console-ng-human-tasks-forms-modeler/jbpm-console-ng-human-tasks-forms-modeler-backend/src/main/java/org/jbpm/console/ng/ht/forms/backend/server/impl/FormRenderContentMarshallerManagerImpl.java]: 2016

File: jbpm-console-ng-human-tasks-forms-modeler/jbpm-console-ng-human-tasks-forms-modeler-backend/src/main/java/org/jbpm/console/ng/ht/forms/backend/server/provider/FormModelerFormProvider.java [hunk 6]
```diff
@@ -0,0 +1,128 @@
+/*
+ * Copyright 2015 Red Hat, Inc. and/or its affiliates.
```
Comment (cristianonicolai) [inline jbpm-console-ng-human-tasks-forms-modeler/jbpm-console-ng-human-tasks-forms-modeler-backend/src/main/java/org/jbpm/console/ng/ht/forms/backend/server/provider/FormModelerFormProvider.java]: 2016

File: jbpm-console-ng-process-runtime-forms/jbpm-console-ng-process-runtime-forms-client/src/main/java/org/jbpm/console/ng/pr/forms/client/display/displayers/process/AbstractStartProcessFormDisplayer.java [hunk 7]
```diff
@@ -132,6 +144,63 @@ public void init(FormDisplayerConfig<ProcessDefinitionKey> config, Command onClo
 
         initDisplayer();
 
+        final PanelGroup accordion = new PanelGroup();
```
Comment (cristianonicolai) [inline jbpm-console-ng-process-runtime-forms/jbpm-console-ng-process-runtime-forms-client/src/main/java/org/jbpm/console/ng/pr/forms/client/display/displayers/process/AbstractStartProcessFormDisplayer.java]: Ideally new UI elements should be done using Errai UI.

File: jbpm-console-ng-workbench-forms-integration/jbpm-console-ng-workbench-forms-integration-api/src/main/resources/org/jbpm/console/ng/workbench/forms/JBPMConsoleNGWorkbenchFormsIntegrationAPI.gwt.xml [hunk 8]
```diff
@@ -0,0 +1,27 @@
+<?xml version="1.0" encoding="utf-8"?>
+
+<!--
```
Comment (cristianonicolai) [inline jbpm-console-ng-workbench-forms-integration/jbpm-console-ng-workbench-forms-integration-api/src/main/resources/org/jbpm/console/ng/workbench/forms/JBPMConsoleNGWorkbenchFormsIntegrationAPI.gwt.xml]: License not needed in here...

File: jbpm-console-ng-workbench-forms-integration/jbpm-console-ng-workbench-forms-integration-client/src/main/resources/META-INF/beans.xml [hunk 9]
```diff
@@ -0,0 +1,22 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!--
+  ~ Copyright 2016 Red Hat, Inc. and/or its affiliates.
```
Comment (cristianonicolai) [inline jbpm-console-ng-workbench-forms-integration/jbpm-console-ng-workbench-forms-integration-client/src/main/resources/META-INF/beans.xml]: License can be removed from *.xml

**Comment by pefernan** [comment]
@cristianonicolai thanks! I'll take a look on Monday.

File: jbpm-console-ng-generic/jbpm-console-ng-generic-forms/jbpm-console-ng-generic-forms-backend/src/main/resources/forms/DefaultProcess.ftl (hunk 2 above)
Comment (pefernan) [inline jbpm-console-ng-generic/jbpm-console-ng-generic-forms/jbpm-console-ng-generic-forms-backend/src/main/resources/forms/DefaultProcess.ftl]: @cristianonicolai agree, this is the template to generate a default form generated when there is no other form defined for a task / process. In a near future there will be a default form generation for the new forms and we could remove it.

File: jbpm-console-ng-process-runtime-forms/jbpm-console-ng-process-runtime-forms-client/src/main/java/org/jbpm/console/ng/pr/forms/client/display/displayers/process/AbstractStartProcessFormDisplayer.java (hunk 7 above)
Comment (pefernan) [inline jbpm-console-ng-process-runtime-forms/jbpm-console-ng-process-runtime-forms-client/src/main/java/org/jbpm/console/ng/pr/forms/client/display/displayers/process/AbstractStartProcessFormDisplayer.java]: @cristianonicolai yeah, that's part of what I told you the other day. If I had more time I would have changed all this part to adapt it to Errai UI.

File: jbpm-console-ng-generic/jbpm-console-ng-generic-forms/jbpm-console-ng-generic-forms-backend/src/main/resources/forms/DefaultProcess.ftl (hunk 2 above)
Comment (cristianonicolai) [inline jbpm-console-ng-generic/jbpm-console-ng-generic-forms/jbpm-console-ng-generic-forms-backend/src/main/resources/forms/DefaultProcess.ftl]: @pefernan that's fine. So perhaps we could simply remove the `<i>` section then?

**Comment by pefernan** [comment]
@cristianonicolai all changes done, can you take a look again?

**Comment by cristianonicolai** [comment]
@pefernan thanks, +1 (once build is successful)

**Comment by psiroky** [comment]
I had to merge this without successful build as it requires different build order of some repositories. Due to the limitation of the Jenkins integration we use, it was not possible to easily run the build with the changed build order. The integration will be fixed to avoid this in the future.

**Pull request merged**